py/torch_tensorrt/dynamo/conversion/converter_utils.py

```diff
@@ -361,12 +361,41 @@ def create_constant(
             shape = list(torch_value.shape)
 
         if torch_value is not None:
+            if torch_value.dtype == torch.float8_e4m3fn:
+                weights = trt.Weights(
+                    type=trt.DataType.FP8,
+                    ptr=torch_value.data_ptr(),
+                    count=torch_value.numel(),
+                )
+                constant = ctx.net.add_constant(
+                    shape,
+                    weights,
+                )
+                constant.name = name
+                ctx.weights_reference_holder[name + " FP8_CONSTANT"] = torch_value
+                return constant.get_output(0)
+
+            if torch_value.dtype == torch.uint8:
+                # Iconstant layer does not support Uint8, it only support that FP4 data packed in uint8
+                shape[-1] = shape[-1] * 2
+                weights = trt.Weights(
+                    type=trt.DataType.FP4,
+                    ptr=torch_value.data_ptr(),
+                    count=torch_value.numel() * 2,
+                )
+                constant = ctx.net.add_constant(
+                    shape,
+                    weights,
+                )
+                constant.name = name
+                ctx.weights_reference_holder[name + " FP4_CONSTANT"] = torch_value
```

Comment:
how do we distinguish a `uint8` constant (stored from FP4 workflow) vs a `uint8` constant manually inserted in the graph. (eg: the graph can have something like `torch.add(x, torch.tensor(2, dtype=torch.uint8))` ?

Reply:
have an optional argument that sets the target type and in the case of uint8 without fp4 then throw error